pkg/cmd/roachtest/operations/changefeeds/utils.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package changefeeds
+
+import (
+	"context"
+	gosql "database/sql"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/operations/helpers"
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+)
+
+// Constants defining environment variables to manage changefeed jobs.
+const (
+	maxChangefeeds          = "OPS_MAX_CF"             // Max number of changefeeds allowed.
+	minChangefeeds          = "OPS_MIN_CF"             // Min number of changefeeds before pausing/cancelling jobs.
+	pctChangefeedsToPause   = "OPS_PCT_PAUSE_CF"       // Percentage of changefeeds to pause.
+	maxPctChangeFeedsScanOn = "OPS_MAX_PCT_CF_SCAN_ON" // Max percentage of changefeeds scan-on state.
+)
+
+// jobDetails contains the job ID and its current status as obtained from the DB.
+type jobDetails struct {
+	jobID   string      // Unique ID of the changefeed job.
+	status  jobs.Status // Current status of the job (e.g., running, paused, etc.).
+	payload struct {    // required payload details for the job
+		options map[string]string // the configured options
+		sinkUri string            // the sink URI
+	}
+}
+
+// updatePayloadForJobs fetches additional changefeed payload details for specific jobs from the database
+// and populates the same in the jobdetails
+func updatePayloadForJobs(ctx context.Context, conn *gosql.DB, jobs []*jobDetails) error {
+	if len(jobs) == 0 {
+		return nil
+	}
+	jobIDs := make([]string, 0)
+	jobIDToDetails := make(map[string]*jobDetails)
+	// Collect all job IDs for query.
+	for i := 0; i < len(jobs); i++ {
+		j := jobs[i]
+		jobIDs = append(jobIDs, j.jobID)
+		jobIDToDetails[j.jobID] = j
+	}
+
+	// Construct SQL query to get changefeed options for specific jobs.
+	stmt := fmt.Sprintf("select job_id, "+
```

Comment:
you can also get the bytes of the jobspb.Payload and unmarshal it into the actual protobuf type and use that, rather than making your own. might be cleaner